source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,202 @@
+To build the virtual machine, you choose the attribute `vm`.
+The `vm` attribute is inhereted from `eval.config.system.build` as defined in the [nixos/default.nix](https://github.com/NixOS/nixpkgs/blob/7c164f4bea71d74d98780ab7be4f9105630a2eba/nixos/default.nix#L19).
+
+:::{admonition} NixOS
+On NixOS you usually have a `$NIX_PATH` environment variable set up. You can use your current version of nixpkgs to build the virtual machine with this simpler command:[^nixosrebuild]
+```shell-session
+nix-build '<nixpkgs/nixos>' -A vm -I nixos-config=./configuration.nix
+```
+:::
+
+</details>
+
+[^nixosrebuild]: On NixOS you can create a virtual machine using `nixos-rebuild build-vm -I nixos-config=./configuration.nix`, which wraps the original command. its usage is mentioned [here](https://nixos.org/manual/nixos/stable/#sec-changing-config).
+
+## Running the virtual machine
+
+The previous command creates a link with the name `result` in the working directory.
+It links to the folder that contains the virtual machine.
+
+```shell-session
+ls -R ./result
+```
+
+```console
+    result:
+    bin  system
+
+    result/bin:
+    run-nixos-vm
+```
+
+To run the virtual machine, execute:
+
+```shell-session
+./result/bin/run-nixos-vm
+```
+
+This command opens a window that shows the boot process of the virtual machine and ends at the `gdm` login screen where you can log in as `alice` with the password `testpw`.
+
+Running the virtual machine will create a `nixos.qcow2` file in the folder from which you start the virtual machine.
+This disk image file contains the dynamic state of the virtual machine.
+It can interfere with debugging as it keeps the state of previous runs, for example the user password.
+You should delete this file when you change the configuration.
+
+# References
+
+- Nix manual: [`nix-build` man page](https://nixos.org/manual/nix/stable/command-ref/nix-build.html).
+- Nix manual: [common cli options](https://nixos.org/manual/nix/stable/command-ref/opt-common.html).
+- Nix manual: [`NIX_PATH` environment variable](https://nixos.org/manual/nix/stable/command-ref/env-common.html#env-NIX_PATH).
+- NixOS Manual: [NixOS Configuration](https://nixos.org/manual/nixos/stable/index.html#ch-configuration).
+- NixOS Manual: [Modules](https://nixos.org/manual/nixos/stable/index.html#sec-writing-modules).
+- NixOS Manual Reference: [Options](https://nixos.org/manual/nixos/stable/options.html).
+- NixOS Manual: [NixOS cli](https://nixos.org/manual/nixos/stable/#sec-changing-config).
+- Wiki entry: [nixos-rebuild build-vm](https://nixos.wiki/wiki/NixOS:nixos-rebuild_build-vm).
+- Source code: [configuration template](https://github.com/NixOS/nixpkgs/blob/b4093a24a868708c06d93e9edf13de0b3228b9c7/nixos/modules/installer/tools/tools.nix#L122-L226).
+- Source code: [vm attribute](https://github.com/NixOS/nixpkgs/blob/master/nixos/default.nix).
```

Comment:
```suggestion
- [Nix manual: `nix-build`](https://nixos.org/manual/nix/stable/command-ref/nix-build.html).
- [Nix manual: Common command-line options](https://nixos.org/manual/nix/stable/command-ref/opt-common.html).
- [Nix manual: `NIX_PATH` environment variable](https://nixos.org/manual/nix/stable/command-ref/env-common.html#env-NIX_PATH).
- [NixOS Manual: NixOS Configuration](https://nixos.org/manual/nixos/stable/index.html#ch-configuration).
- [NixOS Manual: Modules](https://nixos.org/manual/nixos/stable/index.html#sec-writing-modules).
- [NixOS Manual: Configuration Options Reference](https://nixos.org/manual/nixos/stable/options.html).
- [NixOS Manual: NixOS command line interface](https://nixos.org/manual/nixos/stable/#sec-changing-config).
- [NixOS Wiki: `nixos-rebuild build-vm`](https://nixos.wiki/wiki/NixOS:nixos-rebuild_build-vm).
```

I think we don't have to go into depth on the source code. The reference manuals are more important, and that's enough information as it is. I also don't think it's super valuable to look at the implementation at this point and in this context, except for curiosity.

Reply:
I like to keep the links to the source code because those aspects are not documented. they might be better documented in the future.

Reply:
True, agreed. Maybe we can quickly find a reasonable way to reference these things in words, e.g.

```
NixOS source code: `vm` attribute in `default.nix`
```

and similarly for the configuration template: project resource, what it is (human-readable), where it lived (path relative to project root)

Reply:
done